src/ripple/rpc/handlers/NoRippleCheck.cpp

```diff
@@ -0,0 +1,186 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2014 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/rpc/impl/Tuning.h>
+#include <ripple/app/paths/RippleState.h>
+#include <ripple/protocol/TxFlags.h>
+
+namespace ripple {
+
+static void fillTransaction (
+    Json::Value& txArray,
+    RippleAddress const& account,
+    std::uint32_t& sequence,
+    Ledger::ref ledger)
+{
+    txArray["Sequence"] = Json::UInt (sequence++);
+    txArray["Account"] = account.humanAccountID ();
+    txArray["Fee"] = Json::UInt (ledger->scaleFeeLoad (10, false));
+}
+
+// {
+//   account: <account>|<account_public_key>
+//   account_index: <number>        // optional, defaults to 0.
+//   ledger_hash : <ledger>
+//   ledger_index : <ledger_index>
+//   limit: integer                 // optional, number of problems
+//   role: gateway|user             // account role to assume
+//   transactions: true             // optional, reccommend transactions
+// }
+Json::Value doNoRippleCheck (RPC::Context& context)
+{
+    auto const& params (context.params);
+    if (! params.isMember (jss::account))
+        return RPC::missing_field_error ("account");
+
+    if (! params.isMember ("role"))
+        return RPC::missing_field_error ("role");
```

Comment:
In fact, `missing_field_error` looks like it can use the `Json::StaticString` as well.